### PR TITLE
cmd/snap-confine: allow snap-update-ns to poke writable holes in $SNAP

### PR DESCRIPTION
**Pull request by zyga**
This patch expands the apparmor profile for snap-update-ns, when
running as a child of snap-confine, to allow it to construct writable
holes in the $SNAP directory, using the writable mimic construct
introduced in earlier patches.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: cmd/snap-confine/snap-confine.apparmor.in [hunk 1]
```diff
@@ -535,6 +535,30 @@
         /var/snap/{,*/} r,
         /var/snap/*/**/ rw,
 
+        # Allow creating placeholder directory in /tmp/.snap/ as support for
+        # the writable mimic code that can poke holes in arbitrary read-only
+        # places using tmpfs and bind mounts.
+        /tmp/ r,
+        /tmp/** rw,
```
Comment (stolowski) [inline cmd/snap-confine/snap-confine.apparmor.in]: This seems to be more generous than the comment above?

Comment (zyga) [reply cmd/snap-confine/snap-confine.apparmor.in]: Thanks, I will see if we can be more specific.

Comment (mvo5) [reply cmd/snap-confine/snap-confine.apparmor.in]: I agree with Pawel, either the comment needs an update for the rule (the later is of course prefered :)

Comment (zyga) [reply cmd/snap-confine/snap-confine.apparmor.in]: I made the rule more strict now.

File: cmd/snap-confine/snap-confine.apparmor.in [hunk 2]
```diff
@@ -535,6 +535,31 @@
         /var/snap/{,*/} r,
         /var/snap/*/**/ rw,
 
+        # Allow creating placeholder directory in /tmp/.snap/ as support for
+        # the writable mimic code that can poke holes in arbitrary read-only
+        # places using tmpfs and bind mounts.
+        /tmp/ r,
+        /tmp/.snap/ rw,
+        /tmp/.snap/** rw,
```
Comment (jdstrand) [inline cmd/snap-confine/snap-confine.apparmor.in]: Not a blocker: I would probaly rewrite this is:
```
/tmp/.snap/{,**} rw,
```

File: cmd/snap-confine/snap-confine.apparmor.in [hunk 3]
```diff
@@ -535,6 +535,31 @@
...
+        # Allow unmounting the temporary directory in /tmp once it is no longer
+        # necessary.
+        umount /tmp/.snap/**,
+
+        # Allow creating missing directories under $SNAP once the writable
+        # mimic is ready.
+        / r,
+        /snap/ r,
+        /snap/*/ r,
+        /snap/*/** rw,
```
Comment (jdstrand) [inline cmd/snap-confine/snap-confine.apparmor.in]: Not a blocker. I would probably rewrite this as:
```
/snap/{,**} r,
/snap/*/** w,
```

File: cmd/snap-confine/snap-confine.apparmor.in (hunk 2 above)
Comment (zyga) [reply cmd/snap-confine/snap-confine.apparmor.in]: Applied

File: cmd/snap-confine/snap-confine.apparmor.in (hunk 3 above)
Comment (zyga) [reply cmd/snap-confine/snap-confine.apparmor.in]: Applied

**Pull request merged**